### ISSUE TITLE
ActiveRecord::ConnectionAdapters::SQLite3Adapter (FrozenError)

### ISSUE THREAD

**Issue by edariedl** [basecamp/console1984#33]
Hi everyone,

We have recently updated our application to Ruby 3.0.2 and Rails 7.0.0.alpha2 so I wanted to try console1984. I've enabled it in development environment but I am getting `ActiveRecord::ConnectionAdapters::PostgreSQLAdapter (FrozenError)`. 

To make sure it is not caused by our application I created new vanila rails app (ruby 3.0.2 and rails 7.0.0.alpha2) with console1984 but problem stayed the same but with SQLite database.

Here is the stacktrace from test app:
```
[consoletest#main]→ rails c


You have access to production data here. That's a big deal. As part of our promise to keep customer data safe and private, we audit the commands you type here. Let's get started!



Commands:

* decrypt!: enter unprotected mode with access to encrypted information

Unnamed, why are you using this console today?
asdfsadf
/Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract_adapter.rb:79:in `quoted_table_names': can't modify frozen #<Class:ActiveRecord::ConnectionAdapters::SQLite3Adapter>: ActiveRecord::ConnectionAdapters::SQLite3Adapter (FrozenError)
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/sqlite3/quoting.rb:16:in `quote_table_name'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/sqlite3_adapter.rb:386:in `table_structure'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract/schema_statements.rb:116:in `columns'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/schema_cache.rb:117:in `block in columns'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/schema_cache.rb:116:in `fetch'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/schema_cache.rb:116:in `columns'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/schema_cache.rb:125:in `block in columns_hash'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/schema_cache.rb:124:in `fetch'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/schema_cache.rb:124:in `columns_hash'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/model_schema.rb:568:in `load_schema!'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/attributes.rb:264:in `load_schema!'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/encryption/encryptable_record.rb:124:in `load_schema!'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/model_schema.rb:554:in `block in load_schema'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/model_schema.rb:551:in `synchronize'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/model_schema.rb:551:in `load_schema'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/model_schema.rb:417:in `attribute_types'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/model_schema.rb:443:in `type_for_attribute'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/type_caster/map.rb:16:in `type_for_attribute'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/arel/table.rb:107:in `type_for_attribute'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/table_metadata.rb:18:in `type'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/predicate_builder.rb:59:in `build'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/predicate_builder.rb:54:in `[]'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/predicate_builder.rb:126:in `block in expand_from_hash'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/predicate_builder.rb:79:in `each'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/predicate_builder.rb:79:in `flat_map'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/predicate_builder.rb:79:in `expand_from_hash'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/predicate_builder.rb:25:in `build_from_hash'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/query_methods.rb:1277:in `build_where_clause'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/query_methods.rb:720:in `where!'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/query_methods.rb:715:in `where'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation/finder_methods.rb:81:in `find_by'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/relation.rb:182:in `find_or_create_by!'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0.alpha2/lib/active_record/querying.rb:22:in `find_or_create_by!'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/console1984-0.1.15/lib/console1984/sessions_logger/database.rb:9:in `block in start_session'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_silence.rb:18:in `block in silence'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_thread_safe_level.rb:44:in `log_at'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_silence.rb:18:in `silence'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger.rb:64:in `block (3 levels) in broadcast'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_silence.rb:18:in `block in silence'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_thread_safe_level.rb:44:in `log_at'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_silence.rb:18:in `silence'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger.rb:62:in `block (2 levels) in broadcast'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/console1984-0.1.15/lib/console1984/sessions_logger/database.rb:56:in `block in silence_logging'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_silence.rb:18:in `block in silence'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_thread_safe_level.rb:44:in `log_at'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_silence.rb:18:in `silence'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger.rb:64:in `block (3 levels) in broadcast'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_silence.rb:18:in `block in silence'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_thread_safe_level.rb:44:in `log_at'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger_silence.rb:18:in `silence'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/activesupport-7.0.0.alpha2/lib/active_support/logger.rb:62:in `block (2 levels) in broadcast'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/console1984-0.1.15/lib/console1984/sessions_logger/database.rb:55:in `silence_logging'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/console1984-0.1.15/lib/console1984/sessions_logger/database.rb:8:in `start_session'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/console1984-0.1.15/lib/console1984/supervisor.rb:57:in `start_session'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/console1984-0.1.15/lib/console1984/supervisor.rb:25:in `start'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/console1984-0.1.15/lib/console1984/engine.rb:24:in `block in <class:Engine>'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/railtie.rb:250:in `block in run_console_blocks'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/railtie.rb:275:in `each'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/railtie.rb:275:in `each_registered_block'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/railtie.rb:250:in `run_console_blocks'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/application.rb:522:in `block in run_console_blocks'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/engine/railties.rb:15:in `each'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/engine/railties.rb:15:in `each'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/application.rb:522:in `run_console_blocks'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/engine.rb:449:in `load_console'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/commands/console/console_command.rb:35:in `initialize'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/commands/console/console_command.rb:19:in `new'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/commands/console/console_command.rb:19:in `start'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/commands/console/console_command.rb:102:in `perform'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/command/base.rb:87:in `perform'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/command.rb:48:in `invoke'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/railties-7.0.0.alpha2/lib/rails/commands.rb:18:in `<main>'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/bootsnap-1.9.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/bootsnap-1.9.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/bootsnap-1.9.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/bootsnap-1.9.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/edariedl/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/bootsnap-1.9.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from bin/rails:4:in `<main>'
```

Everything works fine without console1984. Is there something I might have done wrong or it is possible it might be some weird bug?

**Comment by jorgemanrubia** [comment]
Hi @edariedl I'll look into this one shortly, thanks for the report!

**Comment by jorgemanrubia** [comment]
I just released version 0.1.18 with a fix for this. Thanks again for the report @edariedl!
